### PR TITLE
[Go] production server

### PR DESCRIPTION
**Pull request by jba**
Implement NewFlowServeMux, which returns an http.ServeMux
that has a route for each defined flow.

Provide convenience functions for using that ServeMux as a server,
and also for selecting which server to use based on the current
development environment.

### CODE REVIEW

File: go/genkit/servers.go [hunk 1]
```diff
@@ -280,6 +229,189 @@ type listFlowStatesResult struct {
 	ContinuationToken string       `json:"continuationToken"`
 }
 
+// Copyright 2024 Google LLC
```
Comment (ianlancetaylor) [inline go/genkit/servers.go:232]: What is this copyright message doing down here?

**Comment by jba** [comment]
@pavelgj WDYT of the API here?

 - `StartServer` to start a server depending on `GENKIT_ENV`
 - `StartDevServer` and `StartProdServer` if you know what you want
 - `NewFlowServeMux` to get the prod router piece that has the flow routes, so you can make it part of a larger server.

I know the names don't quite align; do you have suggestions?
(BTW, for the js, I think "flowServer" is better than "flowsServer" because it doesn't have the double 's'.)

**Review by ianlancetaylor — COMMENTED** [review]
This seems to do much more than rename some things.  Not clear on what is happening here.

**Comment by jba** [comment]
I tried to do a poor person's CL stack by making two separate commits. It took the message from the first instead of the second. I edited the top comment.

I meant to suggest reviewing the commits separately.

File: go/genkit/servers.go (hunk 1 above)
Comment (jba) [reply go/genkit/servers.go]: Fixed.

File: go/genkit/servers.go [hunk 2]
```diff
@@ -39,48 +43,39 @@ import (
...
+// StartServer always returns a non-nil error, the one returned by http.ListenAndServe.
+func StartServer(addr string) error {
+	switch e := currentEnvironment(); e {
+	case EnvironmentDev:
+		return StartDevServer(addr)
+	case EnvironmentProd:
+		return StartProdServer(addr)
+	default:
+		return fmt.Errorf("unknown environment value: %q", e)
+	}
```
Comment (pavelgj) [inline go/genkit/servers.go:57]: the idea is that `StartProdServer` should always start (regardless of `currentEnvironment()`), even in dev mode. Dev (reflection API) server should only run in dev mode.

So the logic here should be:

```
if currentEnvironment() == EnvironmentDev {
  StartDevServer(addr)
}
StartProdServer(addr)
```

Comment (jba) [reply go/genkit/servers.go]: Ah, I see. I was going about this wrong.
I removed StartServer and changed StartProdServer to StartFlowServer.
When you call StartFlowServer, you'll get a dev server too.

**Review by ianlancetaylor — APPROVED** [review]

**Pull request merged**